tests/test_cleaneval.py

```diff
@@ -0,0 +1,486 @@
+clean_newspaper3k.__name__ = 'newspaper3k'
+clean_trafilatura_no_fallback.__name__ = 'trafilatura-no-fallback'
+clean_trafilatura_with_fallback.__name__ = 'trafilatura-with-fallback'
+
+
+def cleaneval_from_files(gold_path, raw_path, list_of_libraries = [ lambda x: x ], eval_method=do_cleaneval, opt_latex=True, opt_total=True, opt_noheader=False, opt_summary=True, opt_ascii=False, opt_unlabelled=False):
+    '''
+    gold_path is the path to the directory containing the gold standard files
+    raw_path is the path to the directory containing the raw files
+    '''
+    n_processed = {} 
+    raw_files = os.listdir(raw_path)
+    n_files = len(raw_files)
+    eval_list = {}
+    sum = {}
+    ss = {}
+    for library in list_of_libraries:
+        sum[library.__name__] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        ss[library.__name__] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        n_processed[library.__name__] = 0
+    for filename in raw_files:
+        raw_file = raw_path + '/' + filename
+        gold_standard_file = gold_path + '/' + filename
+        if not os.access(gold_standard_file, os.R_OK):
+            continue  # original and cleaned files don't always match in CleanEval gold standard
+        raw_html = slurp_file(raw_file)
+        gold_html = normalize(slurp_file(gold_standard_file), opt_ascii, opt_unlabelled)
+        for library in list_of_libraries:
+            cleaned_html = library(raw_html)
+            eval_list[library.__name__] = eval_method(gold_html, cleaned_html)
+            if not opt_total:
+                print(raw_file+ " ("+library.__name__+")" + "\t" + ("%.2f\t%.2f\t%.2f\t%.2f\t%.2f\t%.2f\t%d\t%d\t%d\t%d\t%d\t%d" % eval_list[library.__name__]))
+            for i, x in enumerate(eval_list[library.__name__]):
+                sum[library.__name__][i] += x
+                ss[library.__name__][i] += x ** 2
+            n_processed[library.__name__] += 1
+    for library in list_of_libraries:
+        if n_processed[library.__name__] < n_files:
+            print("Warning: %d files skipped for %s" % (n_files - n_processed[library.__name__], library.__name__))
+    if not opt_latex:
+        print("Library\t\t\tF\tP\tR\tF.tag\tP.tag\tR.tag\tTP\tFP\tFN\tTP.tag\tFP.tag\tFN.tag")
+    else:
+        print(r'\begin{tabular}{ |c|c|c|c|c|c|c|c|c|c|c|c|c| }')
+        print(r'\hline')
+        print(r'Library & F & P & R & F.tag & P.tag & R.tag & TP & FP & FN & TP.tag & FP.tag & FN.tag \\')
+        print(r'\hline')
+    for library in list_of_libraries:
+        tp = sum[library.__name__][6]
+        fp = sum[library.__name__][7]
+        fn = sum[library.__name__][8]
+        tag_tp = sum[library.__name__][9]
+        tag_fp = sum[library.__name__][10]
+        tag_fn = sum[library.__name__][11]
+        result = __evaluate(tp, fp, fn, tag_tp, tag_fp, tag_fn)
+        if not opt_latex:
+            print(library.__name__ + "\t\t" + ("%.2f\t%.2f\t%.2f\t%.2f\t%.2f\t%.2f\t%d\t%d\t%d\t%d\t%d\t%d\n" % result))
```

Comment:
instead of doing just `library.__name__` here, you can create a simple function that does the transformations you manually did above (like removing the `clean_` from the beginning of the function name)